File: Frontend/src/app/model/taskModel.ts
export class TaskModel {
  taskId: number;
  userId: number;
  title: string;
  status: number;
  startDate?: Date;
  endDate?: Date;
  createdAt: Date;
  updatedAt?: Date;
  content?: string;
  priorityId: number;

  constructor(data: Partial<TaskModel>) {
    this.taskId = data.taskId || 0;
    this.userId = data.userId || 0;
    this.title = data.title || '';
    this.status = data.status || 0;
    this.startDate = data.startDate ? new Date(data.startDate) : undefined;
    this.endDate = data.endDate ? new Date(data.endDate) : undefined;
    this.createdAt = data.createdAt ? new Date(data.createdAt) : new Date();
    this.updatedAt = data.updatedAt ? new Date(data.updatedAt) : undefined;
    this.content = data.content;
    this.priorityId = data.priorityId || 0;
  }
}

File: Middleware/TaskPulse.Domain/Entities/TaskModel.cs
using System.ComponentModel.DataAnnotations;

namespace TaskPulse.Domain.Entities;

public class TaskModel
{
    [Key] 
    [Required] 
    public int? TaskId { get; set; }  

    [Required] 
    public int? UserId { get; set; }  

    [Required] 
    [MaxLength(25)] 
    public string? Title { get; set; } 

    [Required] 
    public byte? Status { get; set; } 

    public DateTime? StartDate { get; set; }  

    public DateTime? EndDate { get; set; }

    [Required] 
    public DateTime? CreatedAt { get; set; } = DateTime.Now; 

    public DateTime? UpdatedAt { get; set; } 

    public string? Content { get; set; }  
    
    public byte? PriorityId { get; set; }  
    
    
}